starter-code/server.js

```diff
@@ -71,20 +72,24 @@ app.post('/articles', function(request, response) {
 });
 
 app.put('/articles/:id', function(request, response) {
-  // TODO: Write a SQL query to update an author record. Remember that our articles now have
+  // DONE: Write a SQL query to update an author record. Remember that our articles now have
   // an author_id property, so we can reference it from the request.body.
-  // TODO: Add the required values from the request as data for the SQL query to interpolate
+  // DONE: Add the required values from the request as data for the SQL query to interpolate
   client.query(
-    ``,
-    []
+    `UPDATE authors
+    SET author=$1, "authorUrl"=$2
+    WHERE author_id=$3;`,
+    [request.body.author, request.body.authorUrl, request.body.author_id]
   )
   .then(function() {
-    // TODO: Write a SQL query to update an article record. Keep in mind that article records
+    // DONE: Write a SQL query to update an article record. Keep in mind that article records
     // now have an author_id, in addition to title, category, publishedOn, and body.
-    // TODO: Add the required values from the request as data for the SQL query to interpolate
+    // DONE: Add the required values from the request as data for the SQL query to interpolate
     client.query(
-      ``,
-      []
+      `UPDATE articles
+      SET title=$1, category=$2, "publishedOn"=$3, body=$4
+      WHERE article_id=$5`,
+      [request.body.title, request.body.category, request.body.publishedOn, request.body.body, request.body.article_id]
```

Comment:
2 things: `author_id` and `article_id`
First, You are missing `author_id`.
Each article needs to have a `author_id`.
Second, `article_id` is not passed in within request body, so you can't technically get it from request body.

If you wonder what the request looks like...
You can check out `article.js` line 71.

```
Article.prototype.updateRecord = function(callback) {
  $.ajax({
    url: `/articles/${this.article_id}`,
    method: 'PUT',
    data: {
      author: this.author,
      authorUrl: this.authorUrl,
      body: this.body,
      category: this.category,
      publishedOn: this.publishedOn,
      title: this.title,
      author_id: this.author_id
    }
  })
  .then(function(data) {
    console.log(data);
    if (callback) callback();
  });
};
```

And from here, you know that `article_id` is passed in as an endpoint like
on line 74 `app.put('/articles/:id', function(request, response) {`
that `id` is exactly what we need.

Now, look at how we access that article_id in delete method (`server.js` line 97 )